airgun/widgets.py

```diff
@@ -489,39 +513,58 @@ class CustomParameter(Widget):
         //textarea[@placeholder='Value']
 
     """
-    add_new_value = Text(".//a[contains(text(),'+ Add Parameter')]")
-    new_parameter_name = TextInput(
-        locator=".//input[@placeholder='Name' and not(@value)]")
-    new_parameter_value = TextInput(
-        locator=".//textarea[@placeholder='Value' and not(text())]")
-    NAMES = (
-        ".//tr[contains(@id, 'os_parameter')]/td/input[@placeholder='Name']")
-    VALUE = (
-        ".//table[contains(@id, 'parameters')]//tr"
-        "/td[input[contains(@id, 'name')][contains(@value, '{}')]]"
-        "/following-sibling::td//textarea"
-    )
+    add_new_value = Text("..//a[contains(text(),'+ Add Parameter')]")
+
+    def __init__(self, *args, **kwargs):
```

Comment:
```
def __init__(self, parent, locator=None, id=None, logger=None):
        """Supports initialization via ``locator=`` or ``id=``"""
        if locator and id or not locator and not id:
            raise TypeError('Please specify either locator or id')
```
approach looks more readable and better for understanding. And you will not need these pops